### ISSUE TITLE
Push notifications are not received when the app is in the background in the browser

### ISSUE THREAD

**Issue by ntamas** [phonegap/phonegap-plugin-push#2259]
### Expected Behaviour

Push notifications should be received in the app (in the registered `notification` handler) when running in the browser and the tab of the app is closed or is in the background.

### Actual Behaviour

No `notification` events are received; development console in Firefox complains about `messageChannel` being `undefined` in the service worker code (`TypeError: messageChannel is undefined`).

#### Steps to Reproduce 

1. Open a browser-enabled Cordova app with push notifications in the browser.
2. Put it in the background; wait for the service worker to stop. (In Firefox 58, you can check whether it's running with `about:debugging` and then navigating to the "Workers" tab).
3. When the service worker has stopped, send a push notification to the app.
4. Go back to the tab of the app and open the development console; you should see `TypeError: messageChannel is undefined`)

#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)

Browser platform, tested with Firefox 58

#### Cordova CLI version and cordova platform version

Cordova 8.0.0, browser platform 5.0.3

#### Plugin version

2.2.1

#### Sample Push Data Payload

```
{
    "title": "foo",
    "message": "bar"
}
```

Additional info: the issue is kinda expected because the service worker tries to send the push notification immediately to the running app instance, even if the app has not been started and the worker has not received the message channel yet. I think that the solution would be to store notifications temporarily in IndexedDB when the app is not running, and forward the notifications to the app the next time it sends the message channel to the service worker.

I am happy to work on this in the form of a PR if you agree that this is the right way to go.

**Comment by macdonst** [comment]
@ntamas yeah, feel free to send a PR on this. For the most part I've only been using the browser implementation to test application logic. It would be great if it was more robust.

**Comment by rikenpatel20** [comment]
@ntamas, any luck with this issue? I am facing the same issue in the browser platform.

_(TypeError: messageChannel is undefined)._
